java/src/main/java/com/google/appengine/tools/pipeline/Job.java

```diff
@@ -385,6 +386,15 @@ private void setCurrentRunGuid(String guid) {
     updateSpec.getNonTransactionalGroup().includeSlot(promisedValue.getSlot());
     return promisedValue;
   }
+  
+  public <F> PromisedValue<F> promise(String promiseHandle) {
+    Slot slot = PipelineManager.getPromisedValueSlot(promiseHandle);
+    PromisedValueImpl<F> promisedValue = null;
+    if (slot != null) {
+      promisedValue = new PromisedValueImpl<F>(slot);
+    }
+    return promisedValue;
```

Comment:
should it really return a null if not found? In any case, document.
Also, when adding tests please make sure you cover the case of returning a PromissedValue that was already filled (as well as one that wasn't).
You can have more checks here (e.g. verify that Slot belongs to the same rootJob...)